apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/AbstractSpanAtomicReference.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.impl.transaction;
+
+import javax.annotation.Nullable;
+
+public class AbstractSpanAtomicReference<V extends AbstractSpan<?>> {
```

Comment:
Why is this called `Abstract...` when it's not abstract?

Reply:
AtomicReferenceForAbstractSpan ?

Reply:
Given that `Transaction` _is a_ `Span`, I think `SpanAtomicReference` would be appropriate. Let me know if you object, I can rename it.